backend/src/api/auth/auth.controller.js

```diff
@@ -34,23 +34,30 @@ exports.loginUser = async (req, res, next) => {
   const existingUser = await UserModel.findOne({ email });
 
   if (!existingUser) {
-    return next(new AppError("Email is wrong", 404));
+    return next(new AppError('Email is wrong', 404));
   }
 
   const validPassword = await bcrypt.compare(
     password,
-    existingUser.passwordHash
+    existingUser.passwordHash,
   );
   if (!validPassword) {
-    return next(new AppError("Password is wrong", 403));
+    return next(new AppError('Password is wrong', 403));
   }
-
+  const expiresIn = 2 * 24 * 60 * 60;
   const token = jwt.sign({ id: existingUser._id }, process.env.JWT_SECRET, {
-    expiresIn: 2 * 24 * 60 * 60,
+    expiresIn,
+  });
+  existingUser.tokens.push({ token, expires: Date.now() + expiresIn });
+  const indexesExpiredTokens = [];
+  existingUser.tokens.forEach((token, idx) => {
```

Comment:
First, it could be moved to a separate function for better readability. Second, why not `.filter` method? It would make this code much more clear

Reply:
Reworked